internal/log/log.go

```diff
@@ -20,3 +24,54 @@ func Logger() *logrus.Logger {
 
 // L is a convenience alias to Logger.
 var L = Logger
+
+const (
+	ERRORLEVEL   = 1
+	WARNINGLEVEL = 3
+	INFOLEVEL    = 5
+	DEBUGLEVEL   = 7
+	TRACELEVEL   = 9
+)
+
+func NewBufferSink(buffer *bytes.Buffer) logr.LogSink {
+	return bufferSink{
+		buffer: buffer,
+	}
+}
+
+type bufferSink struct {
+	name   string
+	buffer *bytes.Buffer
+}
+
+func (s bufferSink) Bytes() []byte {
+	return s.buffer.Bytes()
+}
+
+var _ logr.LogSink = bufferSink{}
+
+func (s bufferSink) Enabled(level int) bool {
+	return true
+}
+
+func (s bufferSink) Error(err error, msg string, keysAndValues ...interface{}) {
+	s.buffer.WriteString(fmt.Sprintf("%s %v %s %v\n", s.name, err.Error(), msg, keysAndValues))
+}
+
+func (s bufferSink) Info(level int, msg string, keysAndValues ...interface{}) {
```

Comment:
Super weird that log.Info() takes a level param I guess its a go thing ? Would it be better to have Error, Warn, Info Debug and Trace funcs ?

Reply:
It's kind of a philosophical choice. The idea being... Logs are either two things. Info, or Error. If you think about it, it makes sense. By increasing the level of a log, you are saying "This is probably not something you want to see all the time". Hence, debug and trace. Errors on the whole would always be visible in the log, and probably would not increase the level. Warning is just an alias for Info really.

Reply:
Would tend to think that Error is also Info 🤷 I guess that the difference is you can't turn errors off ? .. Feels more verbose and less clear for the logging functionality consumer (log lib user) .. 

` logger.V(log.TRACELEVEL).Info("unnecessary detail for general scenarios") 
`vs
` logger.Trace("extra detail for execution tracing") 
`

Reply:
but if its a standard i guess its ok.. so will resolve this